.github/workflows/ci.yml

```diff
@@ -12,83 +12,60 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out Pulser
-        uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/checkout@v3
+      - name: Pulser + flake8 setup
+        uses: ./.github/workflows/pulser-setup
         with:
-          python-version: 3.8
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install -e ./pulser-core -e ./pulser-simulation
-          pip install -r requirements.txt
+          extra-packages: flake8
       - name: Lint with flake8
         run: flake8
   black:
     runs-on: ubuntu-latest
     steps:
       - name: Check out Pulser
-        uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/checkout@v3
+      - name: Pulser + black setup
+        uses: ./.github/workflows/pulser-setup
         with:
-          python-version: 3.8
-      - name: Install black
-        run: |
-          python -m pip install --upgrade pip
-          pip install black
-          pip install 'black[jupyter]'
+          extra-packages: black
       - name: Check formatting with black
         run: black --check --diff .
   isort:
     runs-on: ubuntu-latest
     steps:
       - name: Check out Pulser
-        uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/checkout@v3
+      - name: Pulser + isort setup
+        uses: ./.github/workflows/pulser-setup
         with:
-          python-version: 3.8
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install -e ./pulser-core -e ./pulser-simulation
-          pip install -r requirements.txt
+          extra-packages: isort
       - name: Check import sorting with isort
         run: isort --check-only --diff .
   typing:
     runs-on: ubuntu-latest
     steps:
       - name: Check out Pulser
-        uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/checkout@v3
+      - name: Pulser + mypy setup
+        uses: ./.github/workflows/pulser-setup
         with:
-          python-version: 3.8
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install -e ./pulser-core -e ./pulser-simulation
-          pip install -r requirements.txt
+          extra-packages: '''mypy\|pytest'''
       - name: Type check with mypy
         run: mypy
   test:
+    if: github.event_name != 'push'
```

Comment:
Question: Why do we not run test on `push` events ? Not super familiar with all the event types in github actions, so maybe test are automatically triggered through a `merge_request` event or something

Reply:
Just because I wrote a more complete test workflow to run only on `push` to the protected branches, so these would be redundant.